src/Toml/Printer.hs

```diff
@@ -78,14 +80,21 @@ prettyKeyValue i = Text.concat . map kvText . HashMap.toList
     valText (Array a)   = "[" <> Text.intercalate ", " (map valText a) <> "]"
 
     timeText :: DateTime -> Text
-    timeText (Zoned z) = showText z
+    timeText (Zoned z) = showZoneTime z
     timeText (Local l) = showText l
     timeText (Day d)   = showText d
     timeText (Hours h) = showText h
 
     showText :: Show a => a -> Text
     showText = Text.pack . show
 
+    showZoneTime :: ZonedTime -> Text
+    showZoneTime 
+        = Text.pack 
+        . (\(x,y) -> x ++ ":" ++ y) 
+        . (\z -> splitAt (length z - 2) z)
+        . formatTime defaultTimeLocale "%FT%T%Q%z"
```

Comment:
The implementation looks a bit hacky. I understand that there's a reason for that. But I would split the implementation into two functions:

```haskell
showZonedTime t = showZonedDateTime t <> showZonedZone t
```

And then you can perform dirty hacks inside `showZonedZone` function only that contains `formatTime "%z"`.